### DISCUSSION TITLE
How do I make it so when the enemy sings your health drains?

### DISCUSSION THREAD

**Discussion by BrandonHatesAnime** [KadeArchive/Kade-Engine#2337] [Q&A] [answered]
Hi, I've seen some tutorials on this but it's on the OG source code and it didn't work on the Kade Engine one. Can someone tell me how?

**Comment by SuxxorState** [comment, marked as answer]
im still experimenting with this myself so there's a few bugs and whatnot (for example, hold notes act like they do in the base game), but here's a primitive version on how to do it:
Open up PlayState.hx, navigate to around line 3000. (if you cant find the area, use the find command (ctrl + f) and look for this:

							dad.holdTimer = 0;

							if (SONG.needsVoices)
								vocals.volume = 1;
					}
					daNote.active = false;

					daNote.kill();
					notes.remove(daNote, true);
					daNote.destroy();
)
right above `daNote.active = false;` , slip in `health -= 0.03;` (it doesnt HAVE to be 0.03, but thats my recommendation).
now, if you DONT want the enemy to kill you, change the code to an if-then statement, like this-
if (health > 0.1)
    health -= 0.03;

thats it really

**Reply by BrandonHatesAnime** [reply]
How would I do it for a specific character or more

**Reply by SuxxorState** [reply]
then youd do something like in the first image for one character (also yeah, i figured out how to get rid of sustain note drain, add in the `&& !daNote.isSustainNote` in the if-then statement :) )
for multiple characters, you'd tile them like in the second image
here's the code if you wanna play around with it
				switch (SONG.player2)
					{
					case 'gf' | 'bf':
						if (health > 0.05 && !daNote.isSustainNote)
							health -= 0.04;
					}
![tfugffufuty](https://user-images.githubusercontent.com/55562463/134826522-02bc05e8-cdad-43a1-92ba-6d6af2448987.PNG)
![dugsvguds](https://user-images.githubusercontent.com/55562463/134826524-1257685f-5edb-4f0a-9d8d-b0436ab01e07.PNG)

**Reply by BrandonHatesAnime** [reply]
ayy thanks my man
